common.php

```diff
@@ -32,3 +34,28 @@ function startsWith($string, $startString) {
   $len = strlen($startString);
   return (substr($string, 0, $len) === $startString);
 }
+
+function get_users_name() {
+	// Gets the first name of the logged in user
+	if(isset($_SESSION)){
+	  $fname = $_SESSION["current_user"];
+	  return $fname->first_name;
+	}else{
+	  return '';
+	}
+  }
+
+  function auth_loggedin() {
+    if(isset($_SESSION) && array_key_exists("AuthKey", $_SESSION)){ // Session is started and authentication key exists
+      // Get the auth token from session
+      $authtoken = $_SESSION["AuthKey"];
+      if(!Loggedin::find_by_user_digest($authtoken)){ // If user does not exist or authentication key does not match an entry in the loggedin table, logout
+        $_SESSION = array();
+        setcookie(session_name(), '', time() - 30000, '/');
+        session_destroy();
+      }
```

Comment:
Should the user be redirected somewhere after the destruction of a session

Reply:
btw what's the idea behind time()-30000?

Reply:
No, I had a redirect there before, but they should only be redirected in ensure_loggedin. Some pages allow you to view them even if youre not logged in right?

The time() -30000 sets the cookie expire time to sometime in the past, so this terminates the cookie

Reply:
ok in that case then we should have a comment there to briefly describe the logic behind `time() -30000`